src/test/java/org/apache/commons/compress/compressors/gzip/GzipParametersTest.java

```diff
@@ -46,4 +50,33 @@ public void testToString() {
         gzipParameters.setOS(GzipParameters.OS.Z_SYSTEM);
         assertTrue(gzipParameters.toString().contains("Z_SYSTEM"));
     }
+
+    @ParameterizedTest
+    @CsvSource({ //@formatter:off
+        "          , helloworld, true",
+        "          , hello�world, true",
+        "          , hello\0world, false",
+        "ISO-8859-1, helloworld, true",
+        "ISO-8859-1, hello�world, true",
+        "ISO-8859-1, hello\0world, false",
+        "UTF-8     , helloworld, true",
+        "UTF-8     , hello�world, true",
+        "UTF-8     , hello\0world, false",
+        "UTF-16BE  , helloworld, false",
+    })//@formatter:on
+    public void testIllegalCommentOrFileName(String charset, String text, boolean shouldPass) {
```

Comment:
Use `final` where you can.

Reply:
This is a polluting practice that should never be enforced.

There are IDE warnings to catch assignment to method parameters, for those who care; even PMD can be set to warn on AvoidReassigningParameters. There is no justification for polluting the code with final, or generating a warning from a missing final if it is not even assigned anywhere.